classes/SenseiManager.php

```diff
@@ -333,4 +380,120 @@ public static function isSensei(int $user_id, System $system): bool {
+    {
+        $isSensei = false;
+        $sensei_result = $system->db->query("SELECT 1 FROM `sensei` WHERE `sensei_id` = '{$user_id}' AND `is_active` = 1");
+        $result = $system->db->fetch($sensei_result);
+        if ($system->db->last_num_rows > 0) {
+            $isSensei = true;
+        }
+        return $isSensei;
+    }
+
+    public static function getLessonModifier(int $player_stat, int $player_exp, int $sensei_stat, int $sensei_exp, bool $is_specialization): float
+    {
+        $lesson_modifier = 2.0;
+        // First 50% no penalty
+        $stat_ratio_penalty = ($player_stat - ($sensei_stat / 2)) / ($sensei_stat / 2);
+        $lesson_modifier -= max(0, $stat_ratio_penalty);
+        // First 50% no penalty
+        $exp_ratio_penalty = ($player_exp - ($sensei_exp / 2)) / ($sensei_exp / 2);
+        $lesson_modifier -= max(0, $exp_ratio_penalty);
+        // If no benefit, return minimum
+        if ($lesson_modifier <= 1) {
+            return 1;
+        }
+        // If specialization, cap at 2x else 1.75x
+        return $is_specialization ? min(2.0, $lesson_modifier) : min(1.75, $lesson_modifier);
+    }
+
+    public static function getLessonCostForPlayer(User $player, System $system): array {
+        $cost = [];
+        $stat_train_length = 600;
+	    // 56.25% of standard
+	    $stat_long_train_length = $stat_train_length * 4;
+        // 30x length (5 hrs), 12x gains: 40% of standard
+        $stat_extended_train_length = $stat_train_length * 30;
+	    // Forbidden seal trainings boost
+        $stat_long_train_length *= $player->forbidden_seal->long_training_time;
+        $stat_extended_train_length = round($stat_extended_train_length * $player->forbidden_seal->extended_training_time);
+        // Minutes * cost/min * rank
+        $standard_lesson_cost = ($stat_train_length / 60) * SenseiManager::LESSON_COST_PER_MINUTE * $player->rank_num;
+        $long_lesson_cost = ($stat_long_train_length / 60) * SenseiManager::LESSON_COST_PER_MINUTE * $player->rank_num;
+        $extended_lesson_cost = ($stat_extended_train_length / 60) * SenseiManager::LESSON_COST_PER_MINUTE * $player->rank_num;
+        $cost['short'] = $standard_lesson_cost;
+        $cost['long'] = $long_lesson_cost;
+        $cost['extended'] = $extended_lesson_cost;
+        return $cost;
+    }
+
+    public static function getLessonDurationForPlayer(User $player, System $system): array {
+        $duration = [];
+        $stat_train_length = 600;
+        // 56.25% of standard
+        $stat_long_train_length = $stat_train_length * 4;
+        // 30x length (5 hrs), 12x gains: 40% of standard
+        $stat_extended_train_length = $stat_train_length * 30;
+        // Forbidden seal trainings boost
+        $stat_long_train_length *= $player->forbidden_seal->long_training_time;
+        $stat_extended_train_length = round($stat_extended_train_length * $player->forbidden_seal->extended_training_time);
```

Comment:
same comment, reference from a single source of truth instead of redeclaring here

Reply:
Moved to TrainingManager